### PR TITLE
Update export option types

### PR DESCRIPTION
**Pull request by godrei**
Introduce `SigningStyle` and `Destination` types to better express, what values are expected when building exportOptions.

### CODE REVIEW

File: exportoptions/properties.go [hunk 1]
```diff
@@ -17,16 +17,16 @@ const EmbedOnDemandResourcesAssetPacksInBundleDefault = true
 // ICloudContainerEnvironmentKey ...
 const ICloudContainerEnvironmentKey = "iCloudContainerEnvironment"
 
+// ICloudContainerEnvironment ...
+type ICloudContainerEnvironment string
```
Comment (tothszabi) [inline exportoptions/properties.go:21]: Why is this needed?

File: exportoptions/properties.go [hunk 2]
```diff
@@ -79,6 +79,9 @@ func (manifest Manifest) ToHash() map[string]string {
 // MethodKey ...
 const MethodKey = "method"
 
+// Method ...
+type Method string
```
Comment (tothszabi) [inline exportoptions/properties.go:83]: Same question here.

File: exportoptions/properties.go (hunk 1 above)
Comment (godrei) [reply exportoptions/properties.go:21]: This was moved from just a few lines down, to have the following consistent structure:
1, exportOptionKey
2, exportOptionValueType
3, exportOptionsValues

File: exportoptions/properties.go (hunk 2 above)
Comment (godrei) [reply exportoptions/properties.go:83]: Same answer as here: https://github.com/bitrise-io/go-xcode/pull/249/files#r1793409564

**Review by tothszabi — APPROVED** [review]
